includes/data-stores/class-wc-admin-reports-coupons-data-store.php

```diff
@@ -315,19 +306,21 @@ protected function get_cache_key( $params ) {
 	 *
 	 * @since 3.5.0
 	 * @param int $order_id Order ID.
-	 * @return void
+	 * @return int|bool Returns -1 if order won't be processed, or a boolean indicating processing success.
 	 */
 	public static function sync_order_coupons( $order_id ) {
 		global $wpdb;
 
 		$order = wc_get_order( $order_id );
 		if ( ! $order ) {
-			return;
+			return -1;
```

Comment:
I guess this comment applies to the same logic in the other files too.

Reply:
I don't believe this will result in an endless loop - perhaps one extra run of the sync functions though.

If an order got deleted before all 4 sync functions are called, the result being calculated won't be either `-4` or `4`, which will result in the order sync retry being scheduled. On that next run, all sync methods should return `-1` since the order no longer exists, and a retry won't be scheduled again.

Reply:
Actually, a retry won't get scheduled if the order is deleted - the first line of `schedule_single_order_process()` checks if the given ID is for an order, which will fail if it's been deleted.

Reply:
I could confirm practically that 
1. Order o1 created
2. Order o1 deleted
3. Scheduled action for o1 runs

does not cause issues.